### DISCUSSION TITLE
자바스크립트 import하는 Script 태그 위치에 관하여..

### DISCUSSION THREAD

**Discussion by KJDsports** [KDT4-team6/Paldo#36] [Q&A] [answered]
일반적으로 script 태그를 보통 헤드태그 안에 넣는데, 왜 자바스크립트는 바디태그 안에 들어가고, 
그것도 시작이나 위쪽이 아닌 닫는태그 바로 위에 위치하는 걸까요?

```<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta ame="viewport" content="width=device-width, initial-scale=1.0">
  <link rel="stylesheet" href="style.css">
  <title></title>
</head>

<body>
  <form class="hidden" id="login-form">
    <input
      required
      maxlength="15"
      type="text"
      placeholder="what is it your name?">
    <input type="submit" value="log in"></input>
  </form>
  <h1 id="greeting" class="hidden">hello nico!</h1>
  <script src="app.js"></script>
  <script src="js/clock.js"></script> 

</body>`

**Comment by kjungit** [comment, marked as answer]
html은 문서를 해독할때 위에서 아래로 시작합니다.
head -> body

자바스크립트로 보통 DOM(html의 요소, style)들을 제어하는 코드잖아요!?
실행순서가 위에서 아래로 실행되기 때문에 자바스크립트 문서를 먼저 해독하게 되면 제어할 요소들을 찾을 수가 없겠죠!?
body안에 있는것들이 뭔지 모르는 상태에서 제어가 되기 때문입니다.

그래서 지금 보내주신 코드는 body안에있는 요소들을 모두 읽은 후 DOM조작을 하는 script를 아래에 넣은거라고 보시면 됩니다!

꼭 저렇게 아래로 넣으시지 않아도 됩니다.
알고 계신거처럼 해드태그에 넣어두고 defer를 통해서 스크립트가 실행되는걸 지연시킬 수 있습니다.
지연이 되면 DOM이 준비된 후 자바스크립트가 실행되기 때문에 보통 해드에 스크립트를 추가하게 되면 defer속성과 함께 사용합니다!

**Reply by KJDsports** [reply]
감사합니다 큰 도움이 되었습니다
